server/src/main/java/org/opensearch/extensions/DiscoveryExtension.java

```diff
@@ -40,16 +43,19 @@ public DiscoveryExtension(
         TransportAddress address,
         Map<String, String> attributes,
         Version version,
-        PluginInfo pluginInfo
+        PluginInfo pluginInfo,
+        List<ExtensionDenpendency> dependencies
     ) {
         super(name, id, ephemeralId, hostName, hostAddress, address, attributes, DiscoveryNodeRole.BUILT_IN_ROLES, version);
         this.pluginInfo = pluginInfo;
+        this.dependencies = dependencies;   
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         pluginInfo.writeTo(out);
+        ((DiscoveryNode) dependencies).writeTo(out);
```

Comment:
Dependencies is a `List`, you shouldn't cast it to another class.  I believe you either want `out.writeList(List<? extends Writeable> list)` here, or you want to write an integer of the size and then write each list element.  [See example here](https://github.com/opensearch-project/anomaly-detection/blob/a3e2b994ed347263a8dbaf5beed6bd5ddd6931a4/src/main/java/org/opensearch/ad/transport/ADResultBulkRequest.java#L55-L58)